### DISCUSSION TITLE
Full page-width image on website and table of contents after it

### DISCUSSION THREAD

**Discussion by jpmorr** [quarto-dev/quarto-cli#7648] [Q&A] [answered]
### Description

I'm building a website using Quarto and on one sub-page I'm trying to use a full screen image as the page title rather than text. To achieve this, I've used the following code:

````quarto
---
pagetitle: "Test Page" 
---

::: {.column-screen}
![**Here's a really wide full screen image that shouldn't be affected by the ToC!**](https://placehold.co/4000x1000)
:::


## What is Lorem Ipsum?
Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.

## Where does it come from?
Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32.

The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from "de Finibus Bonorum et Malorum" by Cicero are also reproduced in their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.

## Why do we use it?
It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).

## Where can I get some?
There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.

````

````quarto
project:
  type: website

website:
  title: "testing site"
  back-to-top-navigation: true
  navbar:
    background: primary
    left:
      - text: "Home"
      - text: "Page 2"
      - text: "Page 3"

format:
  html:
    theme: zephyr
    toc: true 
````

This almost works perfectly except for one small thing - the table of contents. When I use a full width image before any page text, the table of contents on the right is between the navbar and the image when the page loads like this:
![image](https://github.com/quarto-dev/quarto-cli/assets/10863809/c9ccff7d-9ca7-4616-8752-a629befeb3fe)

And then hovers over the image as you scroll:
![image](https://github.com/quarto-dev/quarto-cli/assets/10863809/921e9c1c-feb6-44de-a79d-00d9703b1927)

Is there anyway to specify that the table of contents doesn't appear until after the full width image on this page?

Or is is possible to specify a title image for the page in the front matter so that the table of contents loads after it?

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by jpmorr** [reply]
For me the min requirement is as above, but you can see the file in question in full at https://git.ecdf.ed.ac.uk/graned/chops2024/-/blob/main/venue.qmd

I've been having some issue with the repo today so hopefully it's visible now.

You can possibly also visit the website to see the behaviour I mentioned in the initial post.

Thanks for your time!

**Reply by mcanouil** [reply]
> For me the min requirement is as above,

It is not the "min requirement" to reproduce your issue. For instance, there is not even a table of content.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/62bccaff-dbf7-4252-9fe5-b6d513618310">

We do not need a full website, we need a small reproducible example.
You could use online images such as <https://placehold.co/>, etc.

**Reply by jpmorr** [reply]
OK, so I've updated the original post with the project `yml` file and the page that causes the issue. Hopefully that should be clearer now.

Thanks again.

**Reply by mcanouil** [reply]
You should use `body-header` for this I believe, see <https://quarto.org/docs/websites/website-tools.html>.

**Reply by jpmorr** [reply]
`body-header` doesn't work in this case as it has to go in the `_quarto.yml` which applies it to every page, and not just the one specific page. Also, it places the content below the title if it's being used.

Is it possible to pass markdown content in as the title? That may push the toc below it all the time?

**Reply by mcanouil** [reply]
Yes (in theory), but not sure of the results😅

```yaml
pagetitle: The name to display in tabs, etc. of your browser since title is now ugly
title: |
  whatever you want
```

**Reply by jpmorr** [reply]
Passing the image in as the `title` works - but it doesn't actually solve the problem of the ToC being above the image on page loading and then overlaying it until you are passed the image.

Having thought about this quite a bit now, I guess what I'm trying to achieve is the `title-block-banner` effect from posts but just with an image instead of all the text. I want to add something between the `navbar` and the `body`.

Is is possible to do something like that?

**Reply by mcanouil** [reply]
You could use css tricks and redefine the title bar using the HTML partials.
There are few discussions about this you can find quite easily.

**Reply by jpmorr** [reply]
For example, if I do `title-block-banner: true` then I get an extra coloured bar around the title. If i try to place an image there `title-block-banner: "/files/img/banner.jpg"`, the image isn't used behind the title text, just the colour.

**Reply by jpmorr** [reply]
Ok, I was hoping it was simpler than creating a partial - I'll have to have a bit of a dig to understand how to use html partials it seems.

**Reply by mcanouil** [reply]
> For example, if I do `title-block-banner: true` then I get an extra coloured bar around the title. If i try to place an image there `title-block-banner: "/files/img/banner.jpg"`, the image isn't used behind the title text, just the colour.
> 
> 
> 
> 

Of course it does not work.
It is a Boolean as described in the documentation.

**Reply by jpmorr** [reply]
Actually the following works pretty close to what I want to do

```
---
pagetitle: "Test Page" 
title: |
  ::: {.column-screen}
  ![](https://placehold.co/4000x1000)
  :::
title-block-banner: true
---

# Here's a really wide full screen image that shouldn't be affected by the ToC!
```
 It's just a case of the image always being a little smaller than the title-banner background

**Reply by jpmorr** [reply]
The documentation says an image can be used: https://quarto.org/docs/authoring/title-blocks.html#custom-backgrounds

**Reply by mcanouil** [reply]
My bad, it does.

Then the issue is your path: `title-block-banner: "/files/img/banner.jpg"`. This means at the root of your project there is `files/img/banner.jpg`, but is it the case?

**Reply by mcanouil** [reply]
> It's just a case of the image always being a little smaller than the title-banner background

Again, I am afraid you'll have to learn and play with CSS which can handle those cases.

When you want intermediate to advanced HTML stuff, there is no way around getting it done using HTML/CSS.

**Reply by jpmorr** [reply]
The image is there because if I put the path into `[]()` the image renders. But it must be related to the path as you say.

**Reply by mcanouil** [reply]
See other discussion very much related to this one:
- #1986

**Reply by jpmorr** [reply, marked as answer]
So a little bit more testing shows that the amount of the image rendered is dependent on the size of the `title` string. Since I was trying to pass in an title with an empty or small string I would get no title banner. So just pass in a few line breaks and it works perfectly.

This is a hack, but works as I want it to.
```
---
pagetitle: "Test Page" 
title: |
  <br>
  <br>
  <br>
  <br>
title-block-banner: image.jpg
---

# Here's a really wide full screen image that shouldn't be affected by the ToC!
```

The image path is a problem for some reason though. Image in the root works fine, but any other path the image fails to render for some reason.

**Reply by mcanouil** [reply]
Without knowing the structure of your project hard to see what's the issue, but it works at the root, it must work anywhere within the project (assuming the right path is provided).

**Reply by jpmorr** [reply]
Yes, it was the path: it should have been `title-block-banner: files/img/banner.jpg` and not `title-block-banner: /files/img/banner.jpg`

Strangely both `![](files/img/banner.jpg)` and `![](/files/img/banner.jpg)` will render the image, so I'm not sure why `title-block-banner` doesn't, but resolved.

**Reply by jpmorr** [reply]
Just to note, this hack is not responsive because of the fixed number of line breaks. CSS and partials would be the way to go to make it work properly for all screen sizes.
